app/src/main/java/com/github/geohunt/app/ui/components/challengecreation/ChallengeSettings.kt

```diff
@@ -27,54 +27,59 @@ import java.time.LocalDate
  *  - Challenge difficulty
  *  - Challenge expiration date
  * Takes mutable states of the arguments it has to select
- * @param selectedDifficulty Mutable state of a challenge difficulty that can be modified by user interaction
- * @param selectedDate Mutable state of a LocalDate that can be modified by user interaction.
- *  Can be null representing that there is no expiration date
+ * @param viewModel corresponding view model
  */
 @Composable
-fun ChallengeSettings(selectedDifficulty: MutableState<Challenge.Difficulty>, selectedDate: MutableState<LocalDate?>) {
+fun ChallengeSettings(viewModel: CreateChallengeViewModel) {
     Column(modifier = Modifier
-            .fillMaxWidth()
-            .height(120.dp)
-            .padding(horizontal = 10.dp),
+        .fillMaxWidth()
+        .height(120.dp)
+        .padding(horizontal = 10.dp),
             verticalArrangement = Arrangement.SpaceBetween) {
-        DifficultySelect(selectedDifficulty)
+        DifficultySelect(viewModel)
 
-        DateSelect(selectedDate)
+        DateSelect(viewModel)
     }
 }
 
 @Composable
-fun DifficultySelect(selectedDifficulty: MutableState<Challenge.Difficulty>) {
+private fun DifficultySelect(viewModel: CreateChallengeViewModel) {
```

Comment:
The ViewModel shouldn't be passed like this to subcomponent. Prefer using callbacks.

If in this case it does not create any issue, then it is fine.